scala3doc/src/dotty/dokka/translators/ScalaSignatureProvider.scala

```diff
@@ -98,6 +100,22 @@ object ScalaSignatureProvider:
         val extendPart = builder.text(" extends ").signature(extendType)
         withTypes.foldLeft(extendPart)((bdr, tpe) => bdr.text(" with ").signature(tpe))
 
+  private def givenClassSignature(clazz: DClass, builder: SignatureBuilder): SignatureBuilder =
+    val ext = clazz.get(ClasslikeExtension)
+    val prefixes = builder
+      .modifiersAndVisibility(clazz, "given")
+      .name(clazz.getName, clazz.getDri)
+      .generics(clazz)
+
+    val withGenerics = ext.constructor.toSeq.foldLeft(prefixes){ (bdr, elem) =>
+      bdr.functionParameters(elem)
+    }
+    clazz.kind match
+      case Kind.Given(Some(instance), _) => withGenerics
+        .text(" as ")
+        .signature(instance)
+      case _ => withGenerics
```

Comment:
Shouldn't we have tests for this?

Reply:
It's actually tested by `givenSignatures` as we had these signatures documented, but in different way (we were generating signature from `DFunction` instead of `DClass`).